lib/reline/io/ansi.rb

```diff
@@ -41,7 +41,7 @@ def initialize
   end
 
   def encoding
-    Encoding.default_external
+    @input.external_encoding || Encoding.default_external
```

Comment:
This change fails [one of irb's test](https://github.com/ruby/irb/blob/3da04b97866631ca134026388b226eda92a9549e/test/irb/test_history.rb#L166). Should be fixed by https://github.com/ruby/irb/pull/1022